tensorflow/contrib/tensorrt/resources/trt_allocator.cc

```diff
@@ -37,8 +37,20 @@ void TRTCudaAllocator::free(void* memory) { cudaFree(memory); }
 
 void* TRTDeviceAllocator::allocate(uint64_t size, uint64_t alignment,
                                    uint32_t flags) {
+  // WAR for allocator alignment requirement
+  alignment = 512;
   assert((alignment & (alignment - 1)) == 0);  // zero or a power of 2.
-  void* mem = allocator_->AllocateRaw(alignment, size);
+  void* mem = allocator_->AllocateRaw(alignment, size + alignment);
+  CHECK(mem);
+
+  CHECK(mem_pool_.insert(mem).second);
```

Comment:
Is `mem_pool_` added just for validation purpose? If yes, how about using `mem_map_` to track mappings from aligned address to allocated address regardless whether they're the same (currently `mem_map_` only tracks address pairs whose key and value are different)?

Reply:
Yes, mem_pool_ is only for validation purpose. (maybe not necessary?)
Having an unordered_set aside from unordered_map is trying to save space.

We could: 1) remove the mem_pool_ and validation completely; or 2) if memory consumption is not of concern, we can remove the mem_pool_ and offload everything to mem_map_